app/Collections/Page.php

```diff
@@ -77,16 +77,22 @@ public function get_data( $page ) {
 	}
 
 	public function index_to_typesense() {
-		$page = $_REQUEST['page'] ?? 1;
+		$batch_size      = $_REQUEST['batch_size'] ?? 20;
+		$page            = $_REQUEST['page'] ?? 1;
+		$imported_count  = $_REQUEST['imported_count'] ?? 0;
+		$total_imports   = $_REQUEST['total_imports'] ?? 0;
+		$import_response = array();
+
+		$post_datas = array();
 		if ( 1 == $page ) {
 			$this->initialize();
 		}
 
 		try {
 			$args = [ 
-				'post_type' => [ 'post', 'page' ],
+				'post_type' => [ 'page' ],
 				'post_status' => 'publish',
-				'posts_per_page' => 20,
+				'posts_per_page' => $batch_size,
 				'paged' => $page
 			];
```

Comment:
'update_post_meta_cache' => false: useful when post meta will not be utilized.
'update_post_term_cache' => false: useful when taxonomy terms will not be utilized.

may you add this